### PR TITLE
Allow Linux boards to have a default.parm file in ROMFS

### PR DESCRIPTION
**Pull request by Williangalvani**
This automatically stores default.parm files in that path into romfs, and allows loading the default params from romfs instead of host filesystem.

Tested both loading `@ROMFS/defaults.parm` and  `/defaults.parm`

The idea here is to have custom default parameters for Navigator and other linux boards, embedded into the firmware itself.

Is this the best approach?
Is this there a better place to keep the .parm file?
I mimicked the structure in AP_HAL_ChibiOS, but named the folder "boards" as we don't really have hwdefs for linux.

### CODE REVIEW

File: Tools/ardupilotwaf/boards.py [hunk 1]
```diff
@@ -1000,6 +1000,15 @@ def configure_env(self, cfg, env):
         if cfg.options.apstatedir:
             cfg.define('AP_STATEDIR', cfg.options.apstatedir)
 
+        defaults_file = f'libraries/AP_HAL_Linux/boards/{self.get_name()}/defaults.parm'
```
Comment (khancyr) [inline Tools/ardupilotwaf/boards.py:1003]: unfortunately, I think that we still have the build server on python2 so no f-string

Comment (Williangalvani) [reply Tools/ardupilotwaf/boards.py]: it's been over a year, now... =[

**Review by tridge — APPROVED** [review]
looks good, just need to convince CI!

**Comment by Williangalvani** [comment]
I managed to get CI passing, but I think my logic in fgets is slightly wrong. and it should only return false if both read returns 0 AND the buffer is empty (i ==0), I'll take another look at this tomorrow morning

**Comment by Williangalvani** [comment]
good to go!

**Comment by rmackay9** [comment]
This seems to have stopped sim_vehicle.py from working for me on my WSL l

**Comment by rmackay9** [comment]
The issue appears to be that you must start sim_vehicle from the top directory and can no longer start it from the vehicle directory.  This means the user must always add the "-v ArduCopter" (or whatever).

**Comment by rishabsingh3003** [comment]
Yes, I have the same problem after this PR. I can't build from the top directory like I have been doing for years. Any chance we can fix it?

**Comment by rmackay9** [comment]
@rishabsingh3003 yes, the fix is in this PR https://github.com/ArduPilot/ardupilot/pull/19966.  Just waiting for it to pass CI  and then we will merge it.

**Pull request merged**